pkg/engine/rule/r0008_malicious_ssh_connection.go

```diff
@@ -0,0 +1,183 @@
+package rule
+
+import (
+	"fmt"
+	"log"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/armosec/kubecop/pkg/approfilecache"
+	"github.com/kubescape/kapprofiler/pkg/tracing"
+)
+
+const (
+	R0008ID                             = "R0008"
+	R0008MaliciousSSHConnectionRuleName = "Malicious SSH Connection"
+	MaxTimeDiffInSeconds                = 2
+)
+
+var SSHRelatedFiles = []string{
+	"ssh_config",
+	"sshd_config",
+	"ssh_known_hosts",
+	"ssh_known_hosts2",
+	"ssh_config.d",
+	"sshd_config.d",
+	".ssh",
+	"authorized_keys",
+	"authorized_keys2",
+	"known_hosts",
+	"known_hosts2",
+	"id_rsa",
+	"id_rsa.pub",
+	"id_dsa",
+	"id_dsa.pub",
+	"id_ecdsa",
+	"id_ecdsa.pub",
+	"id_ed25519",
+	"id_ed25519.pub",
+	"id_xmss",
+	"id_xmss.pub",
+}
+
+var R0008MaliciousSSHConnectionRuleDescriptor = RuleDesciptor{
+	ID:          R0008ID,
+	Name:        R0008MaliciousSSHConnectionRuleName,
+	Description: "Detecting ssh connection to disallowed port",
+	Tags:        []string{"ssh", "connection", "port", "malicious"},
+	Priority:    7,
+	Requirements: RuleRequirements{
+		EventTypes:             []tracing.EventType{tracing.OpenEventType, tracing.NetworkEventType},
+		NeedApplicationProfile: false,
+	},
+	RuleCreationFunc: func() Rule {
+		return CreateRuleR0008MaliciousSSHConnection()
+	},
+}
+
+type R0008MaliciousSSHConnection struct {
+	BaseRule
+	accessRelatedFiles        bool
+	sshInitiatorPid           uint32
+	configFileAccessTimeStamp int64
+	allowedPorts              []uint16
+}
+
+type R0008MaliciousSSHConnectionFailure struct {
+	RuleName         string
+	Err              string
+	FixSuggestionMsg string
+	RulePriority     int
+	FailureEvent     *tracing.NetworkEvent
+}
+
+func (rule *R0008MaliciousSSHConnection) Name() string {
+	return R0008MaliciousSSHConnectionRuleName
+}
+
+func CreateRuleR0008MaliciousSSHConnection() *R0008MaliciousSSHConnection {
+	return &R0008MaliciousSSHConnection{accessRelatedFiles: false,
+		sshInitiatorPid:           0,
+		configFileAccessTimeStamp: 0,
+		allowedPorts:              []uint16{22},
+	}
+}
+
+func (rule *R0008MaliciousSSHConnection) SetParameters(params map[string]interface{}) {
+	if params["allowedPorts"] != nil {
+		rule.allowedPorts = params["allowedPorts"].([]uint16)
```

Comment:
```suggestion
		rule.allowedPorts, ok := params["allowedPorts"].([]uint16)
		if !ok{
		 // print error?
		}
```

Reply:
Alternatively - you can tweek the CRD to handle the input checks of the allowed parameters for you

Reply:
done